src/ui/public/courier/data_source/search_source.js

```diff
@@ -150,8 +150,20 @@ export default function SearchSourceFactory(Promise, Private) {
 
     switch (key) {
       case 'filter':
+        let verifiedFilters = val;
+        if (config.get('courier:ignoreFilterIfFieldNotInIndex')) {
+          if (!(val instanceof Array)) val = [val];
+          verifiedFilters = val.filter (function (el) {
+            let keep = true;
+            if ('meta' in el && 'index' in state) {
+              const field = state.index.fields.byName[el.meta.key];
+              if (!field) keep = false;
```

Comment:
Better to return early here, since once you find a missing field, you're done.

```js
if (..) {
  ...
  if (!field) return false;
}
return true;
```